GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/AnonymiseThreeYearOldFeedbackJob.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database.Models;
+using GenderPayGap.Extensions;
+
+namespace GenderPayGap.WebUI.BackgroundJobs.ScheduledJobs
+{
+    public class AnonymiseThreeYearOldFeedbackJob
+    {
+
+        private readonly IDataRepository dataRepository;
+
+        public AnonymiseThreeYearOldFeedbackJob(IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+        public void AnonymiseFeedbackAction()
+        {
+            var runId = JobHelpers.CreateRunId();
+            var startTime = VirtualDateTime.Now;
+            JobHelpers.LogFunctionStart(runId, nameof(AnonymiseFeedbackAction), startTime);
+
+            try
+            {
+                DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+
+                List<Feedback> feedback = dataRepository.GetAll<Feedback>()
+                    .Where(f => DateTime.Compare(f.CreatedDate, threeYearsAgo) <= 0)
+                    .Where(f => !f.HasBeenAnonymised)
+                    .ToList();
+
+                foreach (Feedback feedbackItem in feedback)
+                {
+                    AnonymiseFeedback(feedbackItem);
+                }

```

Comment:
I think it'd be nice to pull this code out into a separate method - just to make it clear that there's two things going on here:
* code specific to anonymising feedback
* generic code about running web jobs

I can't think of a good name for this method though - sometimes we use something like `AnonymiseFeedbackInner` (but happy for you to think of something better!)